checker/src/abstract_value.rs

```diff
@@ -2884,6 +2885,31 @@ impl AbstractValueTrait for Rc<AbstractValue> {
                     default.clone()
                 }
             }
+            Expression::UnknownTagCheck {
+                path,
+                tag,
+                checking_presence,
+            } => {
+                if let Some(val) = environment.value_at(&path) {
+                    let tag_presence = val.has_tag(tag);
```

Comment:
The logic is implemented by the callers of has_tag. I'm thinking about replacing BoolDomain with AbstractValue in tag domain elements for a better control of tag checking.

Reply:
UnknownTagCheck is an expression that refines to either a compile time constant expression or otherwise remains an UnknownTagCheck. By the time this refinement happens, the call to has_tag has long gone and the caller has nothing more to do.

Reply:
By "callers of has_tag" I also mean the refinement routine itself.